code/game/objects/items/weapons/pet_carrier.dm

```diff
@@ -0,0 +1,262 @@
+	if(target.buckled && istype(target.buckled, /obj/effect/energy_net))
+		target.buckled.forceMove(target.loc) // In case someone nets something smaller than normal.
+
+/obj/item/weapon/pet_carrier/relaymove(mob/living/user, direction)
+	if(open)
+		loc.visible_message(SPAN_NOTICE("[user] climbs out of [src]!"), \
+		SPAN_WARNING("[user] jumps out of [src]!"))
+		remove_occupant(user)
+		return
+	else if(!locked)
+		loc.visible_message(SPAN_NOTICE("[user] pushes open the door to [src]!"), \
+		SPAN_WARNING("[user] pushes open the door of [src]!"))
+		open = TRUE
+		//update_appearance()
+		update_icon()
+		return
+	else if(user.client)
+		container_resist(user)
+
+/obj/item/weapon/pet_carrier/container_resist(mob/living/user)
+	//user.changeNext_move(100)
+	//user.last_special = world.time + 100
+	if(user.mob_size <= MOB_SMALL)
+		to_chat(user, SPAN_NOTICE("You poke a limb through [src]'s bars and start fumbling for the lock switch... (This will take some time.)"))
+		to_chat(loc, SPAN_WARNING("You see [user] reach through the bars and fumble for the lock switch!"))
+		if(!do_after(user, rand(300, 400), target = user) || open || !locked || !(user in occupants))
+			return
+		loc.visible_message(SPAN_WARNING("[user] flips the lock switch on [src] by reaching through!"))
+		to_chat(user, FONT_LARGE("Bingo! The lock pops open!"))
+		locked = FALSE
+		playsound(src, 'sound/machines/door/boltsup.ogg', 30, TRUE)
+		//update_appearance()
+		update_icon()
+	else
+		loc.visible_message(SPAN_WARNING("[src] starts rattling as something pushes against the door!"))
+		to_chat(user, SPAN_NOTICE("You start pushing out of [src]... (This will take about 20 seconds.)"))
+		if(!do_after(user, 200, target = user) || open || !locked || !(user in occupants))
+			return
+		loc.visible_message(SPAN_WARNING("[user] shoves out of [src]!"))
+		to_chat(user, SPAN_NOTICE("You shove open [src]'s door against the lock's resistance and fall out!"))
+		locked = FALSE
+		open = TRUE
+		//update_appearance()
+		update_icon()
+		remove_occupant(user)
+
+/obj/item/weapon/pet_carrier/update_icon()
+	if(open)
+		icon_state = initial(icon_state)
+		return ..()
+	icon_state = "[base_icon]_[!occupants.len ? "closed" : "occupied"]"
+	cut_overlays()
+	add_overlay("[base_icon]_[locked ? "" : "un"]locked")
+	return ..()
+/*
+/obj/item/pet_carrier/update_overlays()
+	. = ..()
+	if(!open)
+		. += "[icon_state]_[locked ? "" : "un"]locked"
+// This wont fuckin work, and I don't care to fix it. I'll just make a verb do the same thing.
+/obj/item/pet_carrier/MouseDrop(atom/over_atom)
+	. = ..()
+	var/turf/T = get_turf(over_atom.loc)
+	if(T.check_density(ignore_mobs = TRUE) && usr.CanUseTopic(src, TRUE) && usr.Adjacent(over_atom) && open && occupants.len)
+		usr.visible_message(span_notice("[usr] unloads [src]."), \
+		span_notice("You unload [src] onto [over_atom]."))
+		for(var/V in occupants)
+			remove_occupant(V, T)
+*/
+/obj/item/weapon/pet_carrier/proc/drop_pet()
```

Comment:
this doesn't need to be a /proc/ that's added at initialize unless you intend to remove it later
verbs are intrinsically just what you did here (verbs are the exact same as a proc in byond!) only it basically adds itself automatically (yet can also be removed later)